prometheus-dns-registration-service/main.go

```diff
@@ -0,0 +1,358 @@
+package main
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/kubernetes-sigs/aws-iam-authenticator/pkg/token"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/aws/aws-sdk-go/service/sts/stsiface"
+	"github.com/pkg/errors"
+
+	log "github.com/sirupsen/logrus"
+	yaml "gopkg.in/yaml.v2"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+// This structure is used to decode the configmap
+type config struct {
+	Global struct {
+		EvaluationInterval string `yaml:"evaluation_interval"`
+		ScrapeInterval     string `yaml:"scrape_interval"`
+		ScrapeTimeOut      string `yaml:"scrape_timeout"`
+	} `yaml:"global"`
+	RuleFiles     []string `yaml:"rule_files"`
+	ScrapeConfigs []struct {
+		HonorLabels bool   `yaml:"honor_labels"`
+		JobName     string `yaml:"job_name"`
+		MetricsPath string `yaml:"metrics_path"`
+		Perams      struct {
+			Match []string `yaml:"match[]"`
+		} `yaml:"params"`
+		ScrapeInterval string `yaml:"scrape_interval"`
+		StaticConfigs  []struct {
+			Targets []string `yaml:"targets"`
+		} `yaml:"static_configs"`
+	} `yaml:"scrape_configs"`
+}
+
+type clusterConfig struct {
+	ClusterName              string
+	MasterEndpoint           string
+	CertificateAuthorityData string
+	Session                  *session.Session
+}
+
+type clientConfig struct {
+	Client      *clientcmdapi.Config
+	ClusterName string
+	ContextName string
+	roleARN     string
+	sts         stsiface.STSAPI
+}
+
+func main() {
+	lambda.Start(handler)
+}
+
+func handler() {
+
+	hostedZoneID := os.Getenv("HOSTED_ZONE_ID")
+	clusterName := os.Getenv("CLUSTER_NAME")
+	configMapName := os.Getenv("CONFIG_MAP_NAME")
+
+	log.Info("Getting the k8s client")
+	clientSet, err := k8sClient(clusterName)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Info("Successfully got the k8s client")
+
+	log.Info("Getting Prometheus server configmap")
+	configMap, err := clientSet.CoreV1().ConfigMaps("monitoring").Get(configMapName, metav1.GetOptions{})
+	if err != nil {
+		log.Fatalf("Unable to get the Prometheus server configmap, %v", err)
```

Comment:
nit: logrus has this awesome `.WithError()` method you can use if you want. This could be applied more below.
```suggestion
		log.WithError(err).Fatal("Unable to get the Prometheus server configmap")
```

Reply:
Fixed here and for every other log error message.